smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/tracing/HeaderExtractAdapter.java

```diff
@@ -0,0 +1,21 @@
+package io.smallrye.reactive.messaging.kafka.tracing;
+
+import java.nio.charset.StandardCharsets;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+
+import io.opentelemetry.context.propagation.HttpTextFormat;
+
+public class HeaderExtractAdapter implements HttpTextFormat.Getter<Headers> {
+    public static final HeaderExtractAdapter GETTER = new HeaderExtractAdapter();
+
+    @Override
+    public String get(Headers headers, String key) {
+        final Header header = headers.lastHeader(key);
```

Comment:
should it fail on multiple values? Is it even possible?

Reply:
I think it's specified to only retrieve the last one